### ISSUE TITLE
Create a pandas plot extension for activity-specific dataframes

### ISSUE THREAD

**Issue by aaron-schroeder** [aaron-schroeder/heartandsole#24]
Basically just create a HNS-compatible backend. Not sure whether this would be useful, or the best implementation of plotly/pandas/activity plotting, or even able to accomplish all my desired goals. But I have been wanting to standardize my plotting implementation for a while, because I keep re-implementing something similar in every place I use dataframes with plotly. Maybe a separate repo would be good for encapsulation/testing.

## References
- [Extending Pandas: Plotting backends]
-[Plotly's own pandas backend](https://plotly.com/python/pandas-backend/)
  - Basically a wrapper for plotly express, which doesn't meet all my needs.
